Library/Library.BusinessLogic/Services/BookService.cs

```diff
@@ -39,14 +40,38 @@ public async Task<List<BookDTO>> GetAll()
             return _mapper.Map<List<BookDTO>>(await _bookRepository.List());
         }
 
-        public async Task<List<BookDTO>> SearchByTitle(string title)
+        public async Task<List<BookDTO>> SearchWithCondition(string title, string releaseDate)
         {
-            return _mapper.Map<List<BookDTO>>(await _bookRepository.List(book => book.Title.Equals(title, StringComparison.InvariantCultureIgnoreCase)));
+            DateTime releaseDateTime;
+            bool isDateParsed = DateTime.TryParse(releaseDate, out releaseDateTime);
+
+            Expression<Func<Book, bool>> predicate;
+
+            if (title == null)
+            {
+                predicate = book => book.ReleaseDate == releaseDateTime;
+            }
+            else if (releaseDate == null)
+            {
+                predicate = book => book.Title.ToUpper() == title.ToUpper();
```

Comment:
https://www.godo.dev/csharp/string-comparison-ignore-case/

General info:
https://stackoverflow.com/questions/6120154/what-is-difference-between-different-string-compare-methods
https://docs.microsoft.com/en-us/dotnet/csharp/how-to/compare-strings